ui-admin/src/study/CreateNewCohortModal.tsx

```diff
@@ -0,0 +1,121 @@
+import { useNavContext } from '../navbar/NavContextProvider'
+import React, { useEffect, useState } from 'react'
+import { Portal, PortalStudy } from '@juniper/ui-core'
+import Modal from 'react-bootstrap/Modal'
+import { Button } from '../components/forms/Button'
+import Select from 'react-select'
+import useReactSingleSelect from '../util/react-select-utils'
+import Api, { ExportData } from '../api/api'
+import { useLoadingEffect } from 'api/api-utils'
+import { Field, QueryBuilder } from 'react-querybuilder'
+import 'react-querybuilder/dist/query-builder.scss'
+import { Store } from 'react-notifications-component'
+import { failureNotification } from '../util/notifications'
+
+/**
+ * Returns a cohort builder modal
+ */
+export default function CreateNewCohortModal({ onDismiss }: {onDismiss: () => void}) {
+  const { portalList } = useNavContext()
+  const initialQuery = { combinator: 'and', rules: [] }
+  const [query, setQuery] = useState(initialQuery)
+  const [cohortName, setCohortName] = useState('')
+  const [selectedPortal, setSelectedPortal] = useState<Portal>()
+  const [selectedStudy, setSelectedStudy] = useState<PortalStudy>()
+  const [participantFields, setParticipantFields] = useState<ExportData>()
+
+  useLoadingEffect(async () => {
+    if (!selectedStudy || !selectedPortal) { return }
+    const response = await Api.exportEnrollees(
+      selectedPortal.shortcode,
+      selectedStudy.study.shortcode,
+      'live', { fileFormat: 'JSON', limit: 0 })
+    const result = await response.json()
+    if (!response.ok) {
+      Store.addNotification(failureNotification('Failed to load cohort criteria options', result.message))
+    } else {
+      setParticipantFields(result)
+    }
+  }, [selectedStudy])
```

Comment:
You can just add a 3rd argument to useLoadingEffect to customize the error message, you don't have to check for the error yourself
```suggestion
      setParticipantFields(result)
  }, [selectedStudy], 'failed to load cohort criteria options')
```